### PR TITLE
feat: game create and join pages

### PR DESCRIPTION
**Pull request by ravikp7**

### CODE REVIEW

**Comment by ravikp7** [comment]
feat: game create and join pages

- add dashboard page to create and join game
- add modals for showing and entering game id
- add shortid module for unique id genertion
- fix css centering for components using grid

fixes #40

**Comment by ravikp7** [comment]
`Absolute centering is not there in mobile as we're using semantic UI with pure CSS`
`It'll be fixed if we get other important things done first`
- Fixed responsiveness for this page
screenshot from 2019-03-06 22-41-29
- Responsive Dashboard Page
screenshot from 2019-03-06 22-41-15

**Comment by ravikp7** [comment]
`Modal for Game Creation and Joining`
peek 2019-03-06 22-42

File: client/src/components/common/header/Header.js [hunk 1]
```diff
@@ -6,36 +6,27 @@ import './header.css';
...
     );
   }
 }
 
 Header.propTypes = {
-  children: PropTypes.oneOfType([
-    PropTypes.arrayOf(PropTypes.node),
-    PropTypes.node,
-  ]).isRequired,
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
```
Comment (rishabh1403) [inline client/src/components/common/header/Header.js]: split into lines :)

File: client/src/pages/dashboard/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const [isModalVisible, setModalVisibility] = useState(false);
+  const [modalType, setModalType] = useState('create');
+
+  const showCreateGame = () => {
+    setModalType('create');
+    setModalVisibility(true);
+  };
+
+  const showJoinGame = () => {
+    setModalType('join');
```
Comment (rishabh1403) [inline client/src/pages/dashboard/Dashboard.js:18]: Love This <3

**Comment by ravikp7** [comment]
- Also, there are no style tests for this PR as we're running out of time.
- There's one test for the `Modal Close` button click.

File: client/src/pages/dashboard/modal/InputId.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import './modal-content.css';
+
+const InputId = () => (
+  <div className="modal-content-container">
+    <form action="/game">
```
Comment (rishabh1403) [inline client/src/pages/dashboard/modal/InputId.js]: form actions are not supposed to be used for url change...override on submit event, and take user to `/game/:id` page

File: client/src/pages/dashboard/modal/ShowId.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { generate } from 'shortid';
+import './modal-content.css';
+
+const ShowId = () => {
+  const gameId = generate();
+  return (
+    <div className="modal-content-container">
+      Share this ID with your friend:
+      {' '}
```
Comment (rishabh1403) [inline client/src/pages/dashboard/modal/ShowId.js]: If you want to render a space, you can use `&nbsp;` in html, intent gets more clear....please refer this https://www.rapidtables.com/web/html/html-codes.html

**Review by rishabh1403 — CHANGES_REQUESTED** [review]

File: client/src/pages/dashboard/modal/InputId.js (hunk 3 above)
Comment (ravikp7) [reply client/src/pages/dashboard/modal/InputId.js]: - Thanks for this valuable advice, I'll read more on that later.
- I've used string query as it doesn't need to be cached by the browser.
- Please let me know if it needs to be changed to a param query

File: client/src/components/common/styledButton/StyledButton.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { Button } from 'semantic-ui-react';
 import PropTypes from 'prop-types';
 
 const StyledButton = ({
-  text, url, size, color, link, inverted, fluid,
+  text, url, size, color, link, inverted, fluid, onClick,
```
Comment (nimish-gupta) [inline client/src/components/common/styledButton/StyledButton.js]:
```suggestion
  const StyledButton = ({ url, link , ...props }) => {
    const buttonProps = { to: url, as: link ? Link: Button }
    return <Button {...buttonProps} {...props}>{text}</Button>
  };
```
By above approach you have to set less defaultProps and can use more props of the button

File: client/package.json [hunk 6]
```diff
@@ -10,7 +10,8 @@
     "react-router-dom": "^4.3.1",
     "react-scripts": "2.1.5",
     "semantic-ui-css": "^2.4.1",
-    "semantic-ui-react": "^0.85.0"
+    "semantic-ui-react": "^0.85.0",
+    "shortid": "^2.2.14"
```
Comment (nimish-gupta) [inline client/package.json:14]: Is this more like a random number generator?

**Review by nimish-gupta — COMMENTED** [review]
LGTM

File: client/package.json (hunk 6 above)
Comment (ravikp7) [reply client/package.json:14]: No it isn't. See [here](https://github.com/dylang/shortid) 
@rishabh1403 suggested this one!

**Pull request merged**